proto/frequenz/api/common/v1/metrics.proto

```diff
@@ -22,159 +24,222 @@ message Bounds {
+  // time of aggregation or derivation.
   //
-  // It is important to note that these bounds work together with
-  // `system_exclusion_bounds`.
+  // ### Multiple Bounds
   //
-  // E.g., for the system to accept a charge command,
-  // clients need to request power values within the bounds
-  // `[system_inclusion_bounds.lower, system_exclusion_bounds.lower]`.
-  // This means that clients can only request charge commands with power values
-  // that are within the `system_inclusion_bounds`, but not within
-  // `system_exclusion_bounds`.
-  // Similarly, for the system to accept a discharge command,
-  // clients need to request power values within the bounds
-  // `[system_exclusion_bounds.upper, system_inclusion_bounds.upper]`.
+  // In the case of certain components like batteries, multiple bounds might
+  // exist. These multiple bounds collectively extend the range of allowable
+  // values, effectively forming a union of all given bounds. In such cases,
+  // the value of the metric must be within at least one of the bounds.
+  // In accordance with the passive sign convention, bounds that limit discharge
+  // would have negative numbers, while those limiting charge, such as for the
+  // State of Power (SoP) metric, would be positive. Hence bounds can have
+  // positive and negative values depending on the metric they represent.
   //
-  // The following diagram illustrates the relationship between the bounds.
+  // Example:
+  // The diagram below illustrates the relationship between the bounds.
   // ```
-  //   inclusion.lower                              inclusion.upper
+  //      bound[0].lower                         bound[1].upper
   // <-------|============|------------------|============|--------->
-  //                exclusion.lower    exclusion.upper
+  //                bound[0].upper      bound[1].lower
   // ```
-  // ---- values here are disallowed and wil be rejected
-  // ==== vales here are allowed and will be accepted
-  frequenz.api.common.v1.metrics.Bounds system_inclusion_bounds = 5;
+  // ---- values here are disallowed and will be rejected
+  // ==== values here are allowed and will be accepted
+  repeated Bounds bounds = 6;
+}
+
+// MetricSampleVariant serves as a union type that can encapsulate either a
+// `SimpleMetricSample` or an `AggregatedMetricSample`.
+//
+// This message is designed to offer flexibility in capturing different
+// granularities of metric samples—either a simple single-point measurement
+// or an aggregated set of measurements for a metric.
+//
+// A `MetricSampleVariant`` can hold either a `SimpleMetricSample` or an
+// `AggregatedMetricSample`, but not both simultaneously. Setting one will
+// nullify the other.
+//
+message MetricSampleVariant {
+  oneof metric_return_type {
+    SimpleMetricSample simple_metric = 1;
+    AggregatedMetricSample aggregated_metric = 2;
+  }
+}
+
+// List of supported metrics.
+enum Metric {
+  // Default value.
+  METRIC_UNSPECIFIED = 0;
```

Comment:
Does it make sense to have the `METRIC_` prefix? Just wondering as the enum is already called Metric.

Reply:
It is recommended to prefix protobuf enum variants with enum names. This is because if there are two or more enums in a scope, all their variants share the same scope (not the scope within the enum, but the outer scope to which the enums belong to).

So, we consistently prefix our protobuf enum variants with the enum names. Also, some programming languages support these by default. E.g., `METRIC_UNSPECIFIED` will appear just as `Unspecified` in rust.

Reply:
Got it. Thanks for clarifying!